File: Backend/model/src/main/java/net/chrisparton/sparkled/model/animation/easing/EasingType.java
package net.chrisparton.sparkled.model.animation.easing;

import java.util.Objects;

public class EasingType {

    private EasingTypeCode code;
    private String name;

    public EasingTypeCode getCode() {
        return code;
    }

    public EasingType setCode(EasingTypeCode code) {
        this.code = code;
        return this;
    }

    public String getName() {
        return name;
    }

    public EasingType setName(String name) {
        this.name = name;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof EasingType)) return false;
        EasingType that = (EasingType) o;
        return code == that.code &&
                Objects.equals(name, that.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(code, name);
    }

    @Override
    public String toString() {
        return "EasingType{" +
                "code=" + code +
                ", name='" + name + '\'' +
                '}';
    }
}

File: model/src/main/kotlin/io/sparkled/model/animation/easing/EasingType.kt
package io.sparkled.model.animation.easing

import io.sparkled.model.animation.param.Param
import io.sparkled.model.reference.ReferenceDataItem

data class EasingType(
    override val code: EasingTypeCode = EasingTypeCode.NONE,
    override val name: String = EasingTypeCode.NONE.displayName,
    val params: List<Param> = emptyList()
) : ReferenceDataItem<EasingTypeCode>
